lib/dry/initializer/builder.rb

```diff
@@ -76,10 +76,13 @@ def copy(&block)
     end
 
     def define_readers(mixin)
-      readers = @signature.select { |item| item.settings[:reader] != false }
-                          .map(&:name)
+      define_reader(mixin, :attr_reader, ->(item) { item.settings[:reader] != false })
+      define_reader(mixin, :private, ->(item) { item.settings[:reader] == :private })
```

Comment:
@jmgarnier  You should rebase the branch to the master. I've fixed .travis.yml, so tests should pass